src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -210,9 +220,24 @@ static ImmutableArray<bool> computeDiscards(SeparatedSyntaxList<ParameterSyntax>
 
                 return discardsBuilder.ToImmutableAndFree();
             }
+
+            static void checkAttributes(AnonymousFunctionExpressionSyntax syntax, SyntaxList< AttributeListSyntax> attributeLists, BindingDiagnosticBag diagnostics)
+            {
+                foreach (var attributeList in attributeLists)
+                {
+                    if (syntax.Kind() == SyntaxKind.ParenthesizedLambdaExpression)
+                    {
+                        MessageID.IDS_FeatureLambdaAttributes.CheckFeatureAvailability(diagnostics, attributeList);
+                    }
+                    else
+                    {
+                        Error(diagnostics, ErrorCode.ERR_AttributesNotAllowed, attributeList);
```

Comment:
nit: consider using a dedicated error message suggesting to parenthesize the lambda's parameter list